js_modules/dagit/packages/core/src/pipelines/PipelineRunsRoot.tsx

```diff
@@ -131,8 +135,57 @@ export const PipelineRunsRoot: React.FC<Props> = (props) => {
               );
             }
             const runs = pipelineRunsOrError.results;
+
+            if (!runs.length) {
+              const description = () => {
+                if (!repoAddress) {
+                  return <div>You have not launched any runs for this job.</div>;
+                }
+
+                if (isAssetJob) {
```

Comment:
Should this check go before the `!repoAddress` check?

Reply:
If `isAssetJob` is true, we must have a `repoAddress` available to find the relevant "pipeline" in a loaded repository. So we'll want to check the `!repoAddress` case first, since it implies that we have no information at all about what kind of object the `pipelineName` corresponds to, and can only show a generic empty state.